command/agent/check.go

```diff
@@ -64,7 +68,7 @@ func (c *CheckType) IsTTL() bool {
 
 // IsMonitor checks if this is a Monitor type
 func (c *CheckType) IsMonitor() bool {
-	return c.Script != "" && c.Interval != 0
+	return c.Script != "" && c.DockerContainerId == "" && c.Interval != 0
```

Comment:
Actually this made me realize how sloppy the current `else` is. We should probably add the script check back to Docker and make an explicit check for `IsMonitor` and have an `else` that fails. Right now we will roll into creating a monitor when these requirements aren't met. Sorry for the flip-flop on this one.